buildSrc/src/main/java/net/neoforged/neodev/NeoDevPlugin.java

```diff
@@ -0,0 +1,527 @@
+        // - Jar tasks have special manifest handling, so use Zip.
+        // - The manifest must be the first entry in the jar so LegacyInstaller has to be the first input.
+        var installerJar = tasks.register("installerJar", Zip.class, task -> {
+            task.setGroup(INTERNAL_GROUP);
+            task.getArchiveClassifier().set("installer");
+            task.getArchiveExtension().set("jar");
+            task.setMetadataCharset("UTF-8");
+            task.getDestinationDirectory().convention(project.getExtensions().getByType(BasePluginExtension.class).getLibsDirectory());
+
+            task.from(project.zipTree(project.provider(installerConfig::getSingleFile)), spec -> {
+                spec.exclude("big_logo.png");
+            });
+            task.from(createLauncherProfile.flatMap(CreateLauncherProfile::getLauncherProfile), spec -> {
+                spec.rename(s -> "version.json");
+            });
+            task.from(createInstallerProfile.flatMap(CreateInstallerProfile::getInstallerProfile), spec -> {
+                spec.rename(s -> "install_profile.json");
+            });
+            task.from(project.getRootProject().file("src/main/resources/url.png"));
+            task.from(project.getRootProject().file("src/main/resources/neoforged_logo.png"), spec -> {
+                spec.rename(s -> "big_logo.png");
+            });
+            task.from(createUnixServerArgsFile.flatMap(CreateArgsFile::getArgsFile), spec -> {
+                spec.into("data");
+                spec.rename(s -> "unix_args.txt");
+            });
+            task.from(createWindowsServerArgsFile.flatMap(CreateArgsFile::getArgsFile), spec -> {
+                spec.into("data");
+                spec.rename(s -> "win_args.txt");
+            });
+            task.from(binaryPatchOutputs.binaryPatchesForClient(), spec -> {
+                spec.into("data");
+                spec.rename(s -> "client.lzma");
+            });
+            task.from(binaryPatchOutputs.binaryPatchesForServer(), spec -> {
+                spec.into("data");
+                spec.rename(s -> "server.lzma");
+            });
+            var mavenPath = neoForgeVersion.map(v -> "net/neoforged/neoforge/" + v);
+            task.getInputs().property("mavenPath", mavenPath);
+            task.from(project.getRootProject().files("server_files"), spec -> {
+                spec.into("data");
+                spec.exclude("args.txt");
+                spec.filter(s -> {
+                    return s.replaceAll("@MAVEN_PATH@", mavenPath.get());
+                });
+            });
+
+            // This is true by default (see gradle.properties), and needs to be disabled explicitly when building (see release.yml).
+            if (project.getProperties().containsKey("neogradle.runtime.platform.installer.debug") && Boolean.parseBoolean(project.getProperties().get("neogradle.runtime.platform.installer.debug").toString())) {
```

Comment:
Use `getOrDefault`

Reply:
Doesn't really work because the type of the properties is `Map<String, ?>`. I did move the long constant to a variable though.